File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        Map<Integer, Integer> count = new HashMap<>();
        for(int num: nums){
            count.put(num, count.getOrDefault(num, 0) + 1);
        }
        List<int[]> arr = new ArrayList<>();
        for(Map.Entry<Integer, Integer> entry: count.entrySet()){
            arr.add(new int[] { entry.getKey(), entry.getValue() });
        }
        arr.sort((a,b) -> b[1] - a[1]);

        int[] res = new int[k];
        for(int i=0; i < k; i++){
            res[i] = arr.get(i)[0];
        }
        return res;
    }
}

File: 0347-top-k-frequent-elements/0347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        buckets = [[] for i in range(len(nums) + 1)]
        countN = Counter(nums)

        for num,freq in countN.items():
            buckets[freq].append(num)

        ans = []
        for i in range(len(buckets) - 1, -1, -1):
            for num in buckets[i]:
                ans.append(num)
                k -= 1
                if k == 0:
                    return ans


